zerver/tests/test_send_email.py

```diff
@@ -98,3 +105,67 @@ def test_initialize_connection(self) -> None:
             initialize_connection(backend)
         # 3 more calls to open as we try 3 times before giving up
         self.assertEqual(backend.open.call_count, 6)
+
+    def test_send_email_not_sent(self) -> None:
+        hamlet = self.example_user("hamlet")
+        from_name = FromAddress.security_email_from_name(language="en")
+        # Used to check the output
+        mail = build_email(
+            "zerver/emails/password_reset",
+            to_emails=[hamlet],
+            from_name=from_name,
+            from_address=FromAddress.NOREPLY,
+            language="en",
+        )
+
+        with mock.patch.object(EmailBackend, "send_messages", return_value=0):
```

Comment:
The only difference between these two functions is this line.  Can we parametrize this test, or use a loop inside of it, instead of repeating the ~30 lines of test?

Reply:
Good point. Changes imported.